### ISSUE TITLE
Mac OS 14.2.1   - usb.core.USBError: [Errno 13] Access denied (insufficient permissions)

### ISSUE THREAD

**Issue by teroff** [ma1co/Sony-PMCA-RE#481]
Hello, after several trials I was able to connect my `ZV-E10` and read the camera info
```
Loading app list
Found 11 apps

Using drivers OS-X-MSC, libusb-MTP, libusb-vendor-specific
Looking for Sony devices

Querying mass storage device
Sony DSC is a camera in mass storage mode
```

But when try to run in service shell `./pmca-console.py serviceshell` I get this error
```
Using drivers libusb-MSC, libusb-MTP, libusb-vendor-specific
Looking for Sony devices

Querying mass storage device
Traceback (most recent call last):
  File "/Users/stanislavkozin/Repo/Sony-PMCA-RE/./pmca-console.py", line 104, in <module>
    main()
  File "/Users/stanislavkozin/Repo/Sony-PMCA-RE/./pmca-console.py", line 88, in main
    senserShellCommand()
  File "/Users/stanislavkozin/Repo/Sony-PMCA-RE/pmca/commands/usb.py", line 660, in senserShellCommand
    device = getDevice(driver)
             ^^^^^^^^^^^^^^^^^
  File "/Users/stanislavkozin/Repo/Sony-PMCA-RE/pmca/commands/usb.py", line 222, in getDevice
    devices = list(listDevices(driver))
              ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/stanislavkozin/Repo/Sony-PMCA-RE/pmca/commands/usb.py", line 184, in listDevices
    info = MscDevice(drv).getDeviceInfo()
           ^^^^^^^^^^^^^^
  File "/Users/stanislavkozin/Repo/Sony-PMCA-RE/pmca/usb/__init__.py", line 26, in __init__
    self.reset()
  File "/Users/stanislavkozin/Repo/Sony-PMCA-RE/pmca/usb/__init__.py", line 49, in reset
    super(MscDevice, self).reset()
  File "/Users/stanislavkozin/Repo/Sony-PMCA-RE/pmca/usb/__init__.py", line 29, in reset
    self.driver.reset()
  File "/Users/stanislavkozin/Repo/Sony-PMCA-RE/pmca/usb/driver/generic/__init__.py", line 88, in reset
    self.backend.reset()
  File "/Users/stanislavkozin/Repo/Sony-PMCA-RE/pmca/usb/driver/generic/libusb.py", line 70, in reset
    self.dev.detach_kernel_driver(0)
  File "/usr/local/lib/python3.11/site-packages/usb/core.py", line 1121, in detach_kernel_driver
    self._ctx.backend.detach_kernel_driver(
  File "/usr/local/lib/python3.11/site-packages/usb/backend/libusb1.py", line 920, in detach_kernel_driver
    _check(self.lib.libusb_detach_kernel_driver(dev_handle.handle, intf))
  File "/usr/local/lib/python3.11/site-packages/usb/backend/libusb1.py", line 604, in _check
    raise USBError(_strerror(ret), ret, _libusb_errno[ret])
usb.core.USBError: [Errno 13] Access denied (insufficient permissions)
```

**Comment by teroff** [comment]
So, I found a solution that worked for me, execute the command with `sudo` 
```sudo ./pmca-console.py serviceshell``` and everything worked like a magic, might need to add this to the README file

**Comment by tennom** [comment]
I ran into the similar issue when I ran this code in a Python's virtual environment. Here is how it worked for me on Ubuntu. 
I created Python virtual environment because I don't want to install these dependency packages system wide.
 
Without `sudo`, it shows the problem of permissions. With `sudo`, the terminal uses the default Python from host system, not the virtual environment Python and it can't find the installed packages in virtual environment.
In this case, 
1. after cloning this code, create a python virtual environment in `Sony-PMCA-RE` like `python3 -m venv .venv`. 
1. then `source .venv/bin/activate` to activate the virtual environment and run `which python` to find the Python path.
2. now , use `sudo` and the path from above to run the code like `sudo /YOUR PATH/Sony-PMCA-RE/.venv/bin/python pmca-gui.py`.

This's how it worked for my Sony ILCE-7CM2 to update the firmware and unlock all languages. NOTE: Windows may have slightly different procedures.
